### PR TITLE
app/post_metadata: add a type cast check for OpenGraph images

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
Adds a type cast check, we log the data type found in the `interface{}` for future debugging.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-42867

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by agnivade — COMMENTED** [review]
While this fixes the issue, I'm interested to understand how is this happening. In the code from L259-265

```go
if og != nil {
		return &model.PostEmbed{
			Type: model.PostEmbedOpengraph,
			URL:  firstLink,
			Data: og,
		}, nil
	}
```

We can see that a struct of type `*opengraph.OpenGraph` only gets added to the map. So I am unsure as to how can this happen.

cc @mkraft as well if you have any thoughts on this. We are observing this in our Cloud production.

**Comment by agnivade** [comment]
Hmm .. this is a very specific bug though. The `Type` field was `model.PostEmbedOpengraph` and the `Data` was of type `map[string]interface{}`. So even if the order was jumbled up, I am failing to see a scenario where we end up with something like this. :thinking:

**Comment by mkraft** [comment]
Yeah that's true, I forgot that the value was `map[string]interface{}`. I'm not seeing where that map came from.

File: app/post_metadata.go [hunk 1]
```diff
@@ -295,7 +295,13 @@ func (a *App) getImagesForPost(post *model.Post, imageURLs []string, isNewPost b
 			imageURLs = append(imageURLs, a.getImagesInMessageAttachments(post)...)
 
 		case model.PostEmbedOpengraph:
-			for _, image := range embed.Data.(*opengraph.OpenGraph).Images {
+			openGraph, ok := embed.Data.(*opengraph.OpenGraph)
+			if !ok {
+				mlog.Debug("Could not read the image data: the data could not be casted to OpenGraph",
```
Comment (agnivade) [inline app/post_metadata.go:300]: Should we change this to a warning? Although it's not actionable I feel like this is something admins should be aware of. Any thoughts?

Comment (isacikgoz) [reply app/post_metadata.go:300]: It depends on how frequent this messages would be, considering that, this is not appeared often. `Warning` level could be fine too.

**Comment by isacikgoz** [comment]
> Hmm .. this is a very specific bug though. The `Type` field was `model.PostEmbedOpengraph` and the `Data` was of type `map[string]interface{}`. So even if the order was jumbled up, I am failing to see a scenario where we end up with something like this. 🤔

This could happen if `PostEmbed.Type` set to `opengraph` but PostEmbed.Data` field does not have the structure of `opengraph.OpenGraph` right?

**Comment by agnivade** [comment]
> This could happen if PostEmbed.Type set to opengraph but PostEmbed.Datafield does not have the structure ofopengraph.OpenGraph` right?

Yes, from the error message we can see that it has `map[string]interface{}` rather than `*opengraph.OpenGraph`

**Review by agnivade — APPROVED** [review]

**Pull request merged**